tests/test_handlers.py

```diff
@@ -126,24 +126,21 @@ def test_vt_http_error(self, mock_requests_get, domain_handler, capsys):
         mock_requests_get.return_value = mock_resp
 
         # Thorough test of first _fetch_* method
-        with pytest.raises(SystemExit) as e:
+        with pytest.raises(ErrorAndExit) as e:
             handler._fetch_vt_domain()
 
-        capture = capsys.readouterr()
```

Comment:
Do you think you can write tests for your new method `main.fetch_data`? It would go in [test_cli.py](https://github.com/pirxthepilot/wtfis/blob/main/tests/test_cli.py).

The test should include the capsys checks that have been replaced here (and we want that to make sure the errors do show up in STDERR).

If you're not sure how, that's okay too, I can add in the tests later on.